### PR TITLE
Drop Debug compatibility layers

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Review by yceruto — APPROVED** [review]

**Review by oliverquynh — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
composer.json needs to be updated too, to remove the classmap entry

**Comment by fancyweb** [comment]
@chalasr Line 137 in Console Application.php -> ` && !$phpHandler[0] instanceof LegacyErrorHandler`

**Comment by chalasr** [comment]
@fancyweb fixed, thank you.

@nicolas-grekas if you mean the one added by https://github.com/symfony/symfony/pull/32873#discussion_r309880549, it was reverted there

**Comment by chalasr** [comment]
Removed the `ProxiedMethodReturnExpression` classmap entry

File: composer.json [hunk 1]
```diff
@@ -135,8 +135,7 @@
             "Symfony\\Component\\": "src/Symfony/Component/"
         },
         "classmap": [
-            "src/Symfony/Component/Intl/Resources/stubs",
-            "src/Symfony/Bridge/ProxyManager/Legacy/ProxiedMethodReturnExpression.php"
```
Comment (nicolas-grekas) [inline composer.json]: this should be kept actually
sorry for the misleading comment, the line I had in mind has been removed earlier

File: src/Symfony/Bridge/ProxyManager/Legacy/ProxiedMethodReturnExpression.php [hunk 2]
```diff
@@ -51,9 +51,6 @@ public static function generate(string $returnedValueExpression, ?\ReflectionMet
         if ($functionLoader instanceof ClassLoader) {
             return $functionLoader;
         }
-        if ($functionLoader instanceof \Symfony\Component\Debug\DebugClassLoader) {
-            return $getComposerClassLoader($functionLoader->getClassLoader());
-        }
         if ($functionLoader instanceof \Symfony\Component\ErrorHandler\DebugClassLoader) {
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/ProxyManager/Legacy/ProxiedMethodReturnExpression.php]: we can use a `use` statement in master now

File: composer.json (hunk 1 above)
Comment (chalasr) [reply composer.json]: reverted

File: src/Symfony/Bridge/ProxyManager/Legacy/ProxiedMethodReturnExpression.php (hunk 2 above)
Comment (chalasr) [reply src/Symfony/Bridge/ProxyManager/Legacy/ProxiedMethodReturnExpression.php]: done

File: src/Symfony/Component/Console/Application.php [hunk 3]
```diff
@@ -134,7 +132,7 @@ public function run(InputInterface $input = null, OutputInterface $output = null
         };
         if ($phpHandler = set_exception_handler($renderException)) {
             restore_exception_handler();
-            if (!\is_array($phpHandler) || (!$phpHandler[0] instanceof ErrorHandler && !$phpHandler[0] instanceof LegacyErrorHandler)) {
+            if (!\is_array($phpHandler) || (!$phpHandler[0] instanceof ErrorHandler)) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Console/Application.php]: extra brackets should be removed :)

Comment (chalasr) [reply src/Symfony/Component/Console/Application.php]: right! done

**Comment by nicolas-grekas** [comment]
Thank you @chalasr.

**Pull request merged**